src/midi/midi_mt32.cpp

```diff
@@ -0,0 +1,516 @@
+	        "Digital = 0\n"
+	        "Only digital path is emulated. The output samples correspond to the digital output signal appeared at the DAC entrance.\n"
+	        "Fastest mode.\n\n"
+
+	        "Coarse = 1\n"
+	        "Coarse emulation of LPF circuit. High frequencies are boosted, sample rate remains unchanged.\n"
+	        "A bit better sounding but also a bit slower.\n\n"
+
+	        "Accurate = 2 - default\n"
+	        "Finer emulation of LPF circuit. Output signal is upsampled to 48 kHz to allow emulation of audible mirror spectra above 16 kHz,\n"
+	        "which is passed through the LPF circuit without significant attenuation.\n"
+	        "Sounding is closer to the analog output from real hardware but also slower than the modes 0 and 1.\n\n"
+
+	        "Oversampled = 3\n"
+	        "Same as the default mode 2 but the output signal is 2x oversampled, i.e. the output sample rate is 96 kHz.\n"
+	        "Even slower than all the other modes but better retains highest frequencies while further resampled in DOSBox mixer.");
+
+	const char *mt32reverbModes[] = {"0", "1", "2", "3", "auto", 0};
+	str_prop = sec_prop.Add_string("reverb.mode", when_idle, "auto");
+	str_prop->Set_values(mt32reverbModes);
+	str_prop->Set_help("MT-32 reverb mode");
+
+	const char *mt32reverbTimes[] = {"0", "1", "2", "3", "4",
+	                                 "5", "6", "7", 0};
+	int_prop = sec_prop.Add_int("reverb.time", when_idle, 5);
+	int_prop->Set_values(mt32reverbTimes);
+	int_prop->Set_help("MT-32 reverb decaying time");
+
+	const char *mt32reverbLevels[] = {"0", "1", "2", "3", "4",
+	                                  "5", "6", "7", 0};
+	int_prop = sec_prop.Add_int("reverb.level", when_idle, 3);
+	int_prop->Set_values(mt32reverbLevels);
+	int_prop->Set_help("MT-32 reverb level");
+
+	// Some frequently used option sets
+	const char *rates[] = {"44100", "48000", "32000", "22050", "16000",
+	                       "11025", "8000",  "49716", 0};
+	int_prop = sec_prop.Add_int("rate", when_idle, 44100);
+	int_prop->Set_values(rates);
+	int_prop->Set_help("Sample rate of MT-32 emulation.");
+
+	const char *mt32srcQuality[] = {"0", "1", "2", "3", 0};
+	int_prop = sec_prop.Add_int("src.quality", when_idle, 2);
+	int_prop->Set_values(mt32srcQuality);
+	int_prop->Set_help(
+	        "MT-32 sample rate conversion quality\n"
+	        "Value '0' is for the fastest conversion, value '3' provides for the best conversion quality. Default is 2.");
+
+	bool_prop = sec_prop.Add_bool("niceampramp", when_idle, true);
+	bool_prop->Set_help(
+	        "Toggles \"Nice Amp Ramp\" mode that improves amplitude ramp for sustaining instruments.\n"
+	        "Quick changes of volume or expression on a MIDI channel may result in amp jumps on real hardware.\n"
+	        "When \"Nice Amp Ramp\" mode is enabled, amp changes gradually instead.\n"
+	        "Otherwise, the emulation accuracy is preserved.\n"
+	        "Default is true.");
```

Comment:
I guess we should hide it and see if anyone complains.

Reply:
With threading, we're going to be able to use all of Munt's recommend defaults, which do consume slightly more CPU but give the truest sounding audio quality.